Cards/Targeting/TargetAll.gd

```diff
@@ -0,0 +1,13 @@
+class_name TargetAll extends TargetingBase
+
+func _init() -> void:
+    cast_type = Enums.CardCastType.INSTA_CAST
+    application_type = Enums.ApplicationType.ALL
```

Comment:
Is `application_type` being used anywhere? We still have the `application_type` on `CardBase` that is being used in `can_play_card`, but it doesn't seem like the ones in the targeting functions are being used.

If we want to store `application_type` in the targeting function, then I guess we will need to loop through each effect and consider if the card can be played based on all of the `application_type`

Reply:
I think it's more useful as a way to check that everything is working correctly (that we target the correct thing)
but indeed i don't think I used them explicitly in the targeting

Reply:
I also realized when reading this again that it wouldn't really make sense to go through all the effects to look at the application type.
Take a card that does 3 heal to self and 3 damage to an enemy. The `application_type` of one is `FRIENDLY_ONLY` and the other is `ENEMY_ONLY`, but the card is totally valid